pkg/cmd/docgen/diagrams.go

```diff
@@ -616,12 +616,29 @@ var specs = []stmtSpec{
 		inline: []string{"name_list"},
 	},
 	{
-		name:    "grant_stmt",
-		inline:  []string{"privileges", "privilege_list", "privilege", "targets", "table_pattern_list", "name_list"},
-		replace: map[string]string{"table_pattern": "table_name", "'DATABASE' ( name ( ',' name )* )": "'DATABASE' ( database_name ( ',' database_name )* )", "'TO' ( name ( ',' name )* )": "'TO' ( user_name ( ',' user_name )* )"},
+		name:   "grant_privileges",
+		stmt:   "grant_stmt",
+		inline: []string{"privileges", "privilege_list", "privilege", "targets", "table_pattern_list", "name_list"},
+		replace: map[string]string{
+			"( name | 'CREATE' | 'GRANT' | 'SELECT' )": "( 'CREATE' | 'GRANT' | 'SELECT' | 'DROP' | 'INSERT' | 'DELETE' | 'UPDATE' )",
+			"table_pattern":                            "table_name",
+			"'DATABASE' ( ( name ) ( ( ',' name ) )*":  "'DATABASE' ( ( database_name ) ( ( ',' database_name ) )*",
+			"'TO' ( ( name ) ( ( ',' name ) )*":        "'TO' ( ( user_name ) ( ( ',' user_name ) )*",
+			"| 'GRANT' ( ( ( 'CREATE' | 'GRANT' | 'SELECT' | 'DROP' | 'INSERT' | 'DELETE' | 'UPDATE' ) ) ( ( ',' ( 'CREATE' | 'GRANT' | 'SELECT' | 'DROP' | 'INSERT' | 'DELETE' | 'UPDATE' ) ) )* ) 'TO' ( ( user_name ) ( ( ',' user_name ) )* )": "",
+			"'WITH' 'ADMIN' 'OPTION'": "",
+		},
 		unlink:  []string{"table_name", "database_name", "user_name"},
 		nosplit: true,
 	},
+	{
+		name: "grant_roles",
+		stmt: "grant_stmt",
+		replace: map[string]string{
+			"'GRANT' privileges 'ON' targets 'TO' name_list\n	| ": "",
+			"'GRANT' privilege_list 'TO' name_list": "'GRANT' ( ( ',' role_name ) )* 'TO' ( ( ',' user_name ) )*",
```

Comment:
shouldn't this be `( ( role_name ) ( ( ',' role_name ) )*` (and same for `user_name`)? (or something similar, the parens look odd).

Reply:
Yes, I couldn't figure that out. Thank you!